tests/forte/processors/base/data_augment_replacement_processor_test.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for data augment processors
+"""
+
+import unittest
+import tempfile
+import os
+
+from forte.processors.data_augment.algorithms.text_replacement_op import TextReplacementOp
+from forte.data.selector import AllPackSelector
+from forte.pipeline import Pipeline
+from forte.data.multi_pack import MultiPack
+from forte.data.ontology.top import MultiPackLink
+from forte.data.readers import MultiPackSentenceReader
+from forte.processors.base.data_augment_processor import ReplacementDataAugmentProcessor
+from forte.processors.nltk_processors import NLTKWordTokenizer, NLTKPOSTagger
+from ft.onto.base_ontology import Token, Sentence, Document, Annotation
+
+from ddt import ddt, data, unpack
+
+
+class TmpReplacementDataAugmentProcessor(ReplacementDataAugmentProcessor):
+    def new_pack(self):
+        return MultiPack()
+
+
+class TmpReplacer(TextReplacementOp):
+    def __init__(self, configs={}):
+        super().__init__(configs)
+        self.token_replacement = {
+            "Mary": "Virgin",
+            "noon": "12",
+            "station": "stop",
+            "until": "til",
+        }
+
+    def replace(self, input):
+        return True, self.token_replacement.get(input.text, input.text)
+
+
+@ddt
+class TestReplacementDataAugmentProcessor(unittest.TestCase):
+    def setUp(self):
+        self.test_dir = tempfile.mkdtemp()
+
+    @data((["Mary and Samantha arrived at the bus station early but waited until noon for the bus."],))
+    @unpack
+    def test_replace_token(self, texts):
+        for idx, text in enumerate(texts):
```

Comment:
You don't have to write them to disk if you use `StringReader`: https://github.com/asyml/forte/blob/master/forte/data/readers/string_reader.py